include/SFML/Graphics/StencilMode.hpp

```diff
@@ -0,0 +1,210 @@
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#pragma once
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Graphics/Export.hpp>
+
+
+namespace sf
+{
+
+////////////////////////////////////////////////////////////
+/// \brief Stencil modes for drawing
+///
+////////////////////////////////////////////////////////////
+struct SFML_GRAPHICS_API StencilMode
+{
+    ////////////////////////////////////////////////////////
+    /// \brief Enumeration of the stencil test comparisons that can be performed
+    ///
+    /// The comparisons are mapped directly to their OpenGL equivalents,
+    /// specified by glStencilFunc().
+    ////////////////////////////////////////////////////////
+    enum class Comparison
+    {
+        Never,     //!< The stencil test never passes
+        Less,      //!< The stencil test passes if the new value is less than the value in the stencil buffer
+        LessEqual, //!< The stencil test passes if the new value is less than or equal to the value in the stencil buffer
+        Greater,   //!< The stencil test passes if the new value is greater than the value in the stencil buffer
+        GreaterEqual, //!< The stencil test passes if the new value is greater than or equal to the value in the stencil buffer
+        Equal,        //!< The stencil test passes if the new value is strictly equal to the value in the stencil buffer
+        NotEqual, //!< The stencil test passes if the new value is strictly inequal to the value in the stencil buffer
+        Always    //!< The stencil test always passes
+    };
+
+    ////////////////////////////////////////////////////////
+    /// \brief Enumeration of the stencil buffer update operations
+    ///
+    /// The update operations are mapped directly to their OpenGL equivalents,
+    /// specified by glStencilOp().
+    ////////////////////////////////////////////////////////
+    enum class UpdateOperation
+    {
+        Keep,      //!< If the stencil test passes, the value in the stencil buffer is not modified
+        Zero,      //!< If the stencil test passes, the value in the stencil buffer is set to zero
+        Replace,   //!< If the stencil test passes, the value in the stencil buffer is set to the new value
+        Increment, //!< If the stencil test passes, the value in the stencil buffer is incremented and if required clamped
+        Decrement, //!< If the stencil test passes, the value in the stencil buffer is decremented and if required clamped
+        Invert,    //!< If the stencil test passes, the value in the stencil buffer is bitwise inverted
+    };
```

Comment:
Would it make sense to move this enum class out side the struct, as the access to the enum values is rather long now.

```cpp
sf::StencilMode::Equal
// vs
sf::StencilMode::Comparison::Equal
```

Maybe something like `sf::StencileComparsion` would work.